python_modules/dagster/dagster/_core/definitions/events.py

```diff
@@ -189,6 +190,24 @@ def from_coercible_or_definition(
         else:
             return AssetKey.from_coercible(arg)
 
+    @staticmethod
+    def from_coercible_to_asset_dep(arg: "CoercibleToAssetDep") -> "AssetKey":
+        from dagster._core.definitions.asset_dep import AssetDep
+        from dagster._core.definitions.asset_spec import AssetSpec
+        from dagster._core.definitions.assets import AssetsDefinition
+        from dagster._core.definitions.source_asset import SourceAsset
+
+        if isinstance(arg, AssetsDefinition):
```

Comment:
Need to error on multi_asset with multiple keys https://github.com/dagster-io/dagster/pull/16432#discussion_r1323536425